site/content/docs/user/quick-start.md

```diff
@@ -260,6 +260,16 @@ Currently, kind supports one default way to build a `node-image`
 if you have the [Kubernetes][kubernetes] source in your host machine
 (`$GOPATH/src/k8s.io/kubernetes`), by using `docker`.
 
+To create a cluster from Kubernetes source:
+
+- ensure that Kubernetes is cloned in `$(go env GOPATH)/src/k8s.io/kubernetes`
```

Comment:
noting: this should also be covered in the **NOTE** below, and it may change in the future if Kubernetes stops depending on GOPATH (which is considered an outstanding bug)

Reply:
The duplication is a concern, but I think the README file is harder to discover than the official docs. Maybe the readme should point here instead?

When you say "this should be covered in the NOTE below", do you mean the `NOTE: Building Kubernetes node-images requires...` section? This should be inside that NOTE block? Or the link in the NOTE block should lead to the correct commands? If the former, happy to move this if that is preferred. If that latter, then I think it's still not very clear and warrants duplication here to make it easier for folks to find.

This was prompted after the question coming up a few times in Slack and then after pointing the person to this section in the docs, realizing that it has a lot of information to digest but doesn't actually make it clear what needs to be run to actually produce an image. Basically, I think it would be good to make it easier to discover how to do this so there is less confusion and less of a need for the question to keep being asked in Slack.

If the preference is to not make it so easy in order to force anyone doing this to actually read the docs more thoroughly, then I suppose that makes sense. Just let me know and I can close this out if so.

Reply:
sorry, this was overly terse, and I've been at an even this week and preparing to go OOO for a while now ...

I meant that the note block below currently leads to the docs for building kubernetes.
The correct requirements for building Kubernetes currently require GOPATH.

So the note below should more expansively cover the correct clone location by way of the upstream docs.

I would prefer not to cover kubernetes build requirements in depth in two places, if we have gaps in the upstream docs we should fill them there.

If the concern is that there's one or two tripping points people miss, I think it's OK to cover here, but I'd prefer to keep that to a minimum and not wind up duplicating the document because the upstream document isn't working for some reason.

The further point is: when kubernetes builds stop requiring GOPATH, so will kind node image builds. Duplicating build requirements gives more room to miss updating them since many folks updating Kubernetes docs probably won't look here.

Reply:
I don't think we want to _force_ reading docs more thoroughly, but ideally:
- we effectively call out the need to see the upstream doc for build requirements
- the upstream docs should be clear and easily read
- we should prefer to improve upstream docs where possible

The kubernetes website repo actually has a hard requirement on not duplicating third party content in their style / review guidelines, so for example they've purged all guides to use kind and instead just link to us.

But, I also understand that many kind users are not Kubernetes developers and are perhaps even very new to development and minimizing the reading required is desirable.

On the other hand, this is only _one_ of the requirements to get Kubernetes to build.
IIRC macOS developers on new machines need to install a few things with homebrew like gnu coreutils and an updated bash shell before the build will succeed (besides e.g. Docker).